pkg/cache/task.go

```diff
@@ -494,25 +494,32 @@ func (task *Task) releaseAllocation() {
 			zap.String("task", task.GetTaskState()),
 			zap.String("terminationType", task.terminationType))
 
-		// The message depends on current task state, generate requests accordingly.
-		// If allocated send an AllocationReleaseRequest,
-		// If not allocated yet send an AllocationAskReleaseRequest
+		// Send an AllocationReleaseRequest,
 		var releaseRequest *si.AllocationRequest
 		s := TaskStates()
-		switch task.GetTaskState() {
-		case s.New, s.Pending, s.Scheduling, s.Rejected:
-			releaseRequest = common.CreateReleaseRequestForTask(task.applicationID, task.taskID, task.application.partition, task.terminationType)
-		default:
+
+		// Check if the task is in a state where it has not been allocated yet.
+		if task.GetTaskState() != s.New && task.GetTaskState() != s.Pending &&
+			task.GetTaskState() != s.Scheduling && task.GetTaskState() != s.Rejected {
+
+			// Task is in a state where it might have been allocated.
 			if task.allocationKey == "" {
```

Comment:
One more question, do we need to log empty allocation key for those state now? cc @pbacsko @craigcondit @wilfred-s